python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -269,32 +297,55 @@ def _get_updated_failed_partition_subset(
     partitions_def: PartitionsDefinition,
     current_cached_subset: PartitionsSubset,
     unevaluated_event_records: Sequence[EventLogRecord],
-) -> PartitionsSubset:
+) -> Tuple[PartitionsSubset, Optional[int]]:
     current_failed_partitions = set(current_cached_subset.get_partition_keys())
 
-    incomplete_materialization_runs: Dict[str, str] = {}
+    cursor = None
+    incomplete_materialization_records: Dict[str, EventLogRecord] = {}
     for record in sorted(unevaluated_event_records, key=lambda r: r.storage_id):
         event = check.not_none(record.event_log_entry.dagster_event)
         if event.is_asset_materialization_planned:
             if event.partition:
                 # If we have a planned materilization for a partition, keep track of it to see if we
                 # also find a materialization. If not, we'll check if the run failed and add it to
                 # the failed partitions.
-                incomplete_materialization_runs[event.partition] = record.run_id
+                incomplete_materialization_records[event.partition] = record
```

Comment:
this implementation only keeps track of 1 in progress materialization at a time- the one that started most recently. I think that's fairly reasonable. If 
- run 1 starts before run 2
- run 1 finishes after run 2
- run 1 fails to materialize and run 2 succeeds

what state should the partition be in? This implementation will say materialized, which I think is reasonable. Especially given that we don't know when the actual step that tried to materialize in run 1 failed, it could have happened immediately before run 2 even started.